pgsmo/objects/server/server.py

```diff
@@ -168,19 +163,17 @@ def refresh(self) -> None:
     # IMPLEMENTATION DETAILS ###############################################
     def _fetch_recovery_state(self) -> Dict[str, Optional[bool]]:
         recovery_check_sql = utils.templating.render_template(
-            utils.templating.get_template_path(self.TEMPLATE_ROOT, 'check_recovery.sql', self._conn.version)
+            utils.templating.get_template_path(self.TEMPLATE_ROOT, 'check_recovery.sql', self.version)
         )
 
-        cols, rows = self._conn.execute_dict(recovery_check_sql)
+        cols, rows = self.connection.execute_dict(recovery_check_sql)
```

Comment:
Any reason why we are using a public property and not a state here ?

Reply:
There is no particular reason. I thought that either was fine here, but I will change it back.